{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# LAI checking\n",
    "\n",
    "* use the results from EDTA"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "INTACT_LIST=/output/genomic/plant/Vaccinium/myrtillus/Genome/NCBI.AssemblyAnnotation/01_Repeats/Bilberry.NCBI/Bilberry.NCBI.fa.mod.EDTA.raw/LTR/Bilberry.NCBI.fa.mod.pass.list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "GENOME=/output/genomic/plant/Vaccinium/myrtillus/Genome/Assembly/bilberry_chr01-12_plusUnanchoredSeqs.NCBI.fasta"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "export PATH=/workspace/hraczw/github/programs/LTR_retriever:$PATH"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "The LTR Assembly Index (LAI) is developed to evaluate the assembly continunity of repetitive sequences\n",
      "\n",
      "Usage: ./LAI -genome genome.fa -intact intact.pass.list -all genome.out [options]\n",
      "Options:\n",
      "\t-genome [file]\tThe genome file that is used to generate everything.\n",
      "\t-intact [file]\tA list of intact LTR-RTs generated by LTR_retriever (genome.fa.pass.list).\n",
      "\t-all [file]\tRepeatMasker annotation of all LTR sequences in the genome (genome.fa.out).\n",
      "\t-window [int]\tWindow size for LAI estimation. Default: 3000000 (3 Mb)\n",
      "\t-step [int]\tStep size for the estimation window to move forward. Default: 300000 (300 Kb)\n",
      "\t\t\tSet step size = window size if prefer non-overlapping outputs.\n",
      "\t-q\t\tQuick estimation of LTR identity (much faster for large genomes, may sacrifice ~0.5% of accuracy).\n",
      "\t-qq\t\tNo estimation of LTR identity, only output raw LAI for within species comparison (very quick).\n",
      "\t-mono [file]\tThis parameter is mainly for ployploid genomes. User provides a list of sequence names that represent a monoploid (1x).\n",
      "\t\t\tLAI will calculated only on these sequences if provided. So user can also specify sequence of interest for LAI calculation.\n",
      "\t-iden [0-100]\tMean LTR identity (%) in the monoploid (1x) genome. This parameter will inactivate de novo estimation (same speed to -qq).\n",
      "\t-totLTR [0-100]\tSpecify the total LTR sequence content (%) in the genome instead of estimating from the -all RepeatMasker file.\n",
      "\t-blast [path]\tThe path to the blastn program. If left unspecified, then blastn must be accessible via shell ENV.\n",
      "\t-t [number]\tNumber of threads to run blastn.\n",
      "\t-h\t\tDisplay this help info.\n",
      "\n"
     ]
    },
    {
     "ename": "",
     "evalue": "255",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "LAI -h"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "mkdir: cannot create directory ‘021.LAI_check’: File exists\n"
     ]
    }
   ],
   "source": [
    "mkdir 021.LAI_check\n",
    "WDR=021.LAI_check"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "LIB=/output/genomic/plant/Vaccinium/myrtillus/Genome/NCBI.AssemblyAnnotation/01_Repeats/Bilberry.NCBI/Bilberry.NCBI.fa.mod.EDTA.raw/LTR/Bilberry.NCBI.fa.mod.LTRlib.fa"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">Chr03:32364321..32367610_LTR#LTR/unknown\n",
      "TGACCACAACATAAGAAGATTTAGTACAATTTTATTAACCTTAAACCCTT\n",
      "TCATATTATCCAAAGCAAACTCGTAAGGTAGGGTCGGATACTCGGATGCA\n",
      "TGTAAGTTATCATAAAAGCCAACGGTTATGTCCAATGAGCTTTTTTACGA\n",
      "TAATGTATGCGGTCATTGCTTAAGTAGAAACCATCAATATTCACCAACCA\n",
      "ATAGCAGTTATATGTGTCATTGTATCGACAAAACGCGTCTAACTTCAAAT\n",
      "GGTAAACGATCAAAATTATGTTATTCGAATTCCATTTGAAATTGCACAAG\n",
      "TAGATTGGGACTTCTGTTGTAATGTAGGGATTGAATTCCCAATAGAATAC\n",
      "TTGGCCACTGTTGAGTGTGTGCGTTCCCAGATCTTCATATTTGGAGGAGC\n",
      "AATGAATGGTTAAGGGCGAAGCTTTTGGAACACCGCTGATAATGGTAACC\n"
     ]
    }
   ],
   "source": [
    "head $LIB"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "cp $LIB $WDR"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Currently Loaded Modulefiles:\n",
      "  1) powerPlant/core    4) git/2.21.0         7) asub/2.2\n",
      "  2) texlive/20151117   5) perlbrew/0.76\n",
      "  3) pandoc/1.19.2      6) perl/5.28.0\n"
     ]
    }
   ],
   "source": [
    "module list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Currently Loaded Modulefiles:\n",
      "  1) powerPlant/core    4) git/2.21.0         7) asub/2.2\n",
      "  2) texlive/20151117   5) perlbrew/0.76      8) singularity/3\n",
      "  3) pandoc/1.19.2      6) perl/5.28.0\n"
     ]
    }
   ],
   "source": [
    "module load singularity\n",
    "module list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Bilberry.NCBI.fa.mod.LTRlib.fa\t library.fa.classified\trepeatMasker.out\n",
      "consensi_modified.fa.classified  repeatMasker.err\n"
     ]
    }
   ],
   "source": [
    "ls 021.LAI_check"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [],
   "source": [
    "LIB=$WDR/Bilberry.NCBI.fa.mod.LTRlib.fa"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Job <822649> is submitted to default queue <lowpriority>.\n"
     ]
    }
   ],
   "source": [
    "# run repeatMasker on the annotated LTR fasta file\n",
    "\n",
    "bsub -J repeatMasker -n 32 \\\n",
    "-o $WDR/repeatMasker.out -e $WDR/repeatMasker.err \\\n",
    "\"singularity exec /workspace/hraegb/singularity/repeatmasker-4.1.1.sif RepeatMasker \\\n",
    "-pa 8 -e ncbi -dir $WDR -gff -poly -lib $LIB $GENOME\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Currently Loaded Modulefiles:\n",
      "  1) powerPlant/core    4) git/2.21.0         7) asub/2.2\n",
      "  2) texlive/20151117   5) perlbrew/0.76      8) singularity/3\n",
      "  3) pandoc/1.19.2      6) perl/5.28.0\n"
     ]
    }
   ],
   "source": [
    "module list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Currently Loaded Modulefiles:\n",
      "  1) powerPlant/core     4) git/2.21.0          7) asub/2.2\n",
      "  2) texlive/20151117    5) perlbrew/0.76       8) singularity/3\n",
      "  3) pandoc/1.19.2       6) perl/5.28.0         9) ncbi-blast/2.10.1\n"
     ]
    }
   ],
   "source": [
    "module load ncbi-blast/2.10.1\n",
    "module list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Job <823444> is submitted to default queue <lowpriority>.\n"
     ]
    }
   ],
   "source": [
    "bsub -J LAI -n 36 -o $WDR/LAI.out -e $WDR/LAI.err \\\n",
    "\"LAI \\\n",
    "-genome $GENOME \\\n",
    "-intact $INTACT_LIST \\\n",
    "-all $WDR/bilberry_chr01-12_plusUnanchoredSeqs.NCBI.fasta.out \\\n",
    "-q \\\n",
    "-t 36 \\\n",
    "-totLTR 31.54\""
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Bash",
   "language": "bash",
   "name": "bash"
  },
  "language_info": {
   "codemirror_mode": "shell",
   "file_extension": ".sh",
   "mimetype": "text/x-sh",
   "name": "bash"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
